rustls/src/webpki/mod.rs

```diff
@@ -91,7 +93,18 @@ fn pki_error(error: webpki::Error) -> Error {
             CertRevocationListError::BadSignature.into()
         }
 
+        #[allow(deprecated)]
         RequiredEkuNotFound => CertificateError::InvalidPurpose.into(),
+        RequiredEkuNotFoundContext(webpki::RequiredEkuNotFoundContext { required, present }) => {
```

Comment:
Not an issue being introduced in this PR, but I note that we don't have coverage for either `RequiredEkuNotFound` or `RequiredEkuNotFoundContext`

Reply:
Any suggestion where/how is the best place to add coverage?

Reply:
I think I would start a server using client credentials (ie, `KeyType::get_client_chain()` rather than `KeyType::get_chain()`, ditto for the key), then make a client and call `do_handshake_until_error`.

Reply:
Added a test, which seems to fail despite looking equal...

```
assertion `left == right` failed
  left: InvalidCertificate(InvalidPurposeContext { required: ServerAuth, presented: [ClientAuth] })
 right: InvalidCertificate(InvalidPurposeContext { required: ServerAuth, presented: [ClientAuth] })
```

Reply:
Fixed.